geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/membership/GMSJoinLeave.java

```diff
@@ -2219,10 +2228,11 @@ public void run() {
                 continue;
               }
             } else {
-              if (System.currentTimeMillis() < okayToCreateView) {
+              long timeRemaining = okayToCreateView - System.currentTimeMillis();
```

Comment:
I think using System.currentTimeMillis to measure elapsed time can cause problems because the system clock can change. It would be safer to use System.nanoTime and convert it to millis, or use the new Java 8 Instant and Duration. See: https://www.baeldung.com/java-measure-elapsed-time
You are not introducing the use of currentTimeMillis in this checkin so no need to convert it in the pull request

Reply:
I've opened a ticket to do this in the future